### PR TITLE
feat: make token optional in AWS marketplace, returning a warning instead of validating

### PR DESCRIPTION
**Pull request by royhadad**
1. return a success message when webhook installed with a token
2. return a warning message when webhook is installed without a token
3. early return and display a warning when applying a resource while the webhook does not have a token

Screenshot 2022-12-04 at 16 07 44
Screenshot 2022-12-04 at 16 07 52

### CODE REVIEW

File: charts/datree-admission-webhook-awsmp/values.yaml [hunk 1]
```diff
@@ -22,10 +22,10 @@ rbac:
     create: true
     # The ClusterRole name
     name: datree-webhook-server-read
-# Datree webhook configuration, checkout more details at htttps://hub.datree.com
+# Datree webhook configuration, checkout more details at https://hub.datree.com
 datree:
   # The token used to link the CLI to your dashboard.
-  token: <DATREE_TOKEN>
+  token: <DATREE_TOKEN> # beware when altering the default value "<DATREE_TOKEN>" as it is used throughout the codebase
```
Comment (royhadad) [inline charts/datree-admission-webhook-awsmp/values.yaml:28]: not sure if I should keep this comment, as end users might be exposed to it

File: Makefile [hunk 2]
```diff
@@ -143,7 +143,7 @@ deploy-and-bump-all-local:
 
 
 #########################
-#    DEPLOPY (HELM)     #
+#    DEPLOY (HELM)     #
```
Comment (noaabarki) [inline Makefile:146]: 👍🏼

File: charts/datree-admission-webhook-awsmp/values.yaml (hunk 1 above)
Comment (noaabarki) [reply charts/datree-admission-webhook-awsmp/values.yaml:28]: I think it's good.

**Review by noaabarki — APPROVED** [review]

File: charts/datree-admission-webhook-awsmp/templates/deployment.yaml [hunk 3]
```diff
@@ -85,7 +85,7 @@ spec:
           value: "{{ .Values.debug }}"
         # Datree webhook varaibles
         - name: DATREE_TOKEN
-          value: {{.Values.datree.token | regexFind $uuidv4RegexPattern }}
+          value: {{.Values.datree.token}}
```
Comment (noaabarki) [inline charts/datree-admission-webhook-awsmp/templates/deployment.yaml:88]: I think now we can also delete the variable `uuidv4RegexPattern`

File: pkg/services/validationService.go [hunk 4]
```diff
@@ -88,28 +88,40 @@ func (vs *ValidationService) Validate(admissionReviewReq *admission.AdmissionRev
...
 	if config.IsConfigurationChangeEvent(admissionReviewReq.Request) {
 		vs.configHandler.HandleConfigurationChange(admissionReviewReq.Request)
 	}
 
+	// no token provided - skip validation
+	if token == "<DATREE_TOKEN>" {
+		clusterRequestMetadata := getClusterRequestMetadata(cliEvaluationId, token, false, true, resourceKind, resourceName, managers, clusterK8sVersion, policyName, namespace, server.ConfigMapScanningFilters)
+		saveRequestMetadataLogInAggregator(clusterRequestMetadata)
+
+		const warningMessage1 = `⚠️  Datree is installed, but not activated since you didn't set your token.`
```
Comment (noaabarki) [inline pkg/services/validationService.go:113]: I think it's better to have const for the messages, perhaps with a function that wraps everything, or maybe simply create an array with the values and pass it.

Nevertheless, I don't think it's super important, so it's up to you :)

**Pull request merged**